metadata-io/src/main/java/com/linkedin/metadata/entity/AspectDao.java

```diff
@@ -53,9 +53,10 @@ void saveAspect(
         @Nonnull final Timestamp timestamp,
         @Nonnull final String systemMetadata,
         final long version,
-        final boolean insert);
+        final boolean insert,
+        @Nullable final Long updateIfCreatedOn);
 
-    void saveAspect(@Nonnull final EntityAspect aspect, final boolean insert);
+    void saveAspect(@Nonnull final EntityAspect aspect, final boolean insert, final Long updateIfCreatedOn);
```

Comment:
As we discussed -> Let's add an object instead of `updateIfCreatedOn` so we can make the future extension story simpler :)